client/components/Slide/Components/AudioResponse/Editor.jsx

```diff
@@ -1,48 +1,93 @@
 import {
+    Accordion,
     Checkbox,
     Container,
     Form,
+    Icon,
     Input,
     Message,
     Popup
 } from 'semantic-ui-react';
 import { type } from './type';
+import ResponseRecall from '@components/Slide/Components/ResponseRecall/Editor';
 import './AudioResponse.css';
 
 class AudioResponseEditor extends Component {
     constructor(props) {
         super(props);
-        const { prompt, required, responseId } = props.value;
+        const { prompt, recallId = '', required, responseId } = props.value;
         this.state = {
+            activeIndex: recallId ? 0 : -1,
             prompt,
+            recallId,
             required,
             responseId
         };
         this.onPromptChange = this.onPromptChange.bind(this);
         this.onRequirementChange = this.onRequirementChange.bind(this);
+        this.onRecallChange = this.onRecallChange.bind(this);
+        this.toggleOptional = this.toggleOptional.bind(this);
     }
 
     updateState() {
-        const { prompt, required, responseId } = this.state;
-        this.props.onChange({ prompt, required, responseId, type });
+        const { prompt, recallId, required, responseId } = this.state;
+        this.props.onChange({ prompt, recallId, required, responseId, type });
+    }
+
+    onPromptChange(event, { value }) {
+        this.setState({ prompt: value }, this.updateState);
+    }
+
+    onRecallChange({ recallId }) {
+        this.setState({ recallId }, this.updateState);
     }
 
     onRequirementChange(event, { checked }) {
         this.setState({ required: checked }, this.updateState);
     }
 
-    onPromptChange(event, { value }) {
-        this.setState({ prompt: value }, this.updateState);
+    toggleOptional(event, { index }) {
+        const { activeIndex } = this.state;
+        const newIndex = activeIndex === index ? -1 : index;
+
+        this.setState({ activeIndex: newIndex });
     }
 
     render() {
-        const { prompt, required } = this.state;
-        const { onPromptChange, onRequirementChange } = this;
+        const { activeIndex, prompt, recallId, required } = this.state;
+        const { scenarioId } = this.props;
+        const {
+            onPromptChange,
+            onRecallChange,
+            onRequirementChange,
+            toggleOptional
+        } = this;
         return (
             <Form>
                 <Container fluid>
+                    <Accordion>
+                        <Accordion.Title
+                            active={activeIndex === 0}
+                            index={0}
+                            onClick={toggleOptional}
+                        >
+                            <Icon name="dropdown" />
+                            Optionally Embed A Previous Response
+                        </Accordion.Title>
+                        <Accordion.Content active={activeIndex === 0}>
+                            <ResponseRecall
+                                style={{ marginBottom: '1rem' }}
```

Comment:
non-blocking: can we add this to a CSS file somewhere instead of setting it programmatically?

Reply:
Sure!

Reply:
Done!